addons/setup/addon_setup.go

```diff
@@ -84,17 +84,30 @@ func (a *Addons) Manifests(cluster *clusterv1.ManagedCluster, addon *addonapiv1a
 	groups := agent.DefaultGroups(cluster.Name, a.AddonName)
 	user := agent.DefaultUser(cluster.Name, a.AddonName, a.AddonName)
 
+	var odfOperatorNamespace string
+	if utils.HasRequiredODFKey(cluster) {
+		odfOperatorNamespacedName, err := utils.GetNamespacedNameForClusterInfo(*cluster)
+		if err != nil {
+			return objects, fmt.Errorf("error while getting ODF operator namespace on the spoke cluster %q. %w", cluster.Name, err)
+		}
+		odfOperatorNamespace = odfOperatorNamespacedName.Namespace
+	} else {
+		return objects, fmt.Errorf("error while getting ODF operator namespace on the spoke cluster %q. Expected ClusterClaim does not exist", cluster.Name)
+	}
+
```

Comment:
Why not just use addon install namespace ? We are always creating the addons in ODF's namespace

Reply:
Even for Provider consumer cases, We will be doing the same. Fetching the namespace again would be redundant operation
https://github.com/red-hat-storage/odf-multicluster-orchestrator/pull/221/files#diff-71a31664c2a326d791c6040801a670cc01383300df3e4378f843bb619d2ea4e8R410-R440

Reply:
You are assuming that addon install namespace will be same as ODF operator namespace. That might not always be true. Instead I am allowing a way to change it dynamically.

Reply:
@umanga I feel that fetching ODF namespace does not make sense in the manifests func for setting up addons.

Can we make it so that it is calculated by the MirrorPeer and sent further as a configuration option? 

If there's no time, then we can merge it and design later.